### DISCUSSION TITLE
Prisma 4.3.1 and NodeJS 16.17.0

### DISCUSSION THREAD

**Discussion by ChipClark** [prisma/prisma#15257] [Q&A]
### Bug description

I get an error when I attempt to run

0 verbose cli C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js
1 info using npm@8.15.0
2 info using node@v16.17.0
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 3ms
5 timing config:load:file:C:\Users\tech-cclark\AppData\Roaming\nvm\v16.17.0\node_modules\npm\npmrc Completed in 1ms
6 timing config:load:builtin Completed in 2ms
7 timing config:load:cli Completed in 3ms
8 timing config:load:env Completed in 2ms
9 timing config:load:file:C:\Sites\AM-API-MDD\.npmrc Completed in 0ms
10 timing config:load:project Completed in 3ms
11 timing config:load:file:C:\Users\tech-cclark\.npmrc Completed in 0ms
12 timing config:load:user Completed in 0ms
13 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:credentials Completed in 2ms
17 timing config:load:setEnvs Completed in 2ms
18 timing config:load Completed in 18ms
19 timing npm:load:configload Completed in 19ms
20 timing npm:load:mkdirpcache Completed in 1ms
21 timing npm:load:mkdirplogs Completed in 0ms
22 verbose title npm i prisma@4
23 verbose argv "i" "prisma@4" "--save"
24 timing npm:load:setTitle Completed in 2ms
25 timing config:load:flatten Completed in 5ms
26 timing npm:load:display Completed in 9ms
27 verbose logfile logs-max:10 dir:C:\Users\tech-cclark\AppData\Local\npm-cache\_logs
28 verbose logfile C:\Users\tech-cclark\AppData\Local\npm-cache\_logs\2022-09-07T20_47_04_581Z-debug-0.log
29 timing npm:load:logFile Completed in 12ms
30 timing npm:load:timers Completed in 0ms
31 timing npm:load:configScope Completed in 0ms
32 timing npm:load Completed in 45ms
33 verbose stack C:\Users\tech-cclark\AppData\Roaming\nvm\v16.17.0\node_modules\npm\node_modules\libnpmfund:1
33 verbose stack ../workspaces/libnpmfund
33 verbose stack ^
33 verbose stack
33 verbose stack SyntaxError: Unexpected token '.'
33 verbose stack     at Object.compileFunction (node:vm:360:18)
33 verbose stack     at wrapSafe (node:internal/modules/cjs/loader:1055:15)
33 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1090:27)
33 verbose stack     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
33 verbose stack     at Module.load (node:internal/modules/cjs/loader:1004:32)
33 verbose stack     at Function.Module._load (node:internal/modules/cjs/loader:839:12)
33 verbose stack     at Module.require (node:internal/modules/cjs/loader:1028:19)
33 verbose stack     at require (node:internal/modules/cjs/helpers:102:18)
33 verbose stack     at Object.<anonymous> (C:\Users\tech-cclark\AppData\Roaming\nvm\v16.17.0\node_modules\npm\lib\utils\reify-output.js:16:38)
33 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1126:14)
34 verbose cwd C:\Sites\AM-API-MDD
35 verbose Windows_NT 10.0.17763
36 verbose node v16.17.0
37 verbose npm  v8.15.0
38 error Unexpected token '.'
39 verbose exit 1
40 timing npm Completed in 87ms
41 verbose code 1
42 error A complete log of this run can be found in:
42 error     C:\Users\tech-cclark\AppData\Local\npm-cache\_logs\2022-09-07T20_47_04_581Z-debug-0.log

### How to reproduce

npx generate prisma 

error log: 
```
0 verbose cli C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js
1 info using npm@8.15.0
2 info using node@v16.17.0
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 3ms
5 timing config:load:file:C:\Users\tech-cclark\AppData\Roaming\nvm\v16.17.0\node_modules\npm\npmrc Completed in 1ms
6 timing config:load:builtin Completed in 2ms
7 timing config:load:cli Completed in 3ms
8 timing config:load:env Completed in 2ms
9 timing config:load:file:C:\Sites\AM-API-MDD\.npmrc Completed in 0ms
10 timing config:load:project Completed in 3ms
11 timing config:load:file:C:\Users\tech-cclark\.npmrc Completed in 0ms
12 timing config:load:user Completed in 0ms
13 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:credentials Completed in 2ms
17 timing config:load:setEnvs Completed in 2ms
18 timing config:load Completed in 18ms
19 timing npm:load:configload Completed in 19ms
20 timing npm:load:mkdirpcache Completed in 1ms
21 timing npm:load:mkdirplogs Completed in 0ms
22 verbose title npm i prisma@4
23 verbose argv "i" "prisma@4" "--save"
24 timing npm:load:setTitle Completed in 2ms
25 timing config:load:flatten Completed in 5ms
26 timing npm:load:display Completed in 9ms
27 verbose logfile logs-max:10 dir:C:\Users\tech-cclark\AppData\Local\npm-cache\_logs
28 verbose logfile C:\Users\tech-cclark\AppData\Local\npm-cache\_logs\2022-09-07T20_47_04_581Z-debug-0.log
29 timing npm:load:logFile Completed in 12ms
30 timing npm:load:timers Completed in 0ms
31 timing npm:load:configScope Completed in 0ms
32 timing npm:load Completed in 45ms
33 verbose stack C:\Users\tech-cclark\AppData\Roaming\nvm\v16.17.0\node_modules\npm\node_modules\libnpmfund:1
33 verbose stack ../workspaces/libnpmfund
33 verbose stack ^
33 verbose stack
33 verbose stack SyntaxError: Unexpected token '.'
33 verbose stack     at Object.compileFunction (node:vm:360:18)
33 verbose stack     at wrapSafe (node:internal/modules/cjs/loader:1055:15)
33 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1090:27)
33 verbose stack     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
33 verbose stack     at Module.load (node:internal/modules/cjs/loader:1004:32)
33 verbose stack     at Function.Module._load (node:internal/modules/cjs/loader:839:12)
33 verbose stack     at Module.require (node:internal/modules/cjs/loader:1028:19)
33 verbose stack     at require (node:internal/modules/cjs/helpers:102:18)
33 verbose stack     at Object.<anonymous> (C:\Users\tech-cclark\AppData\Roaming\nvm\v16.17.0\node_modules\npm\lib\utils\reify-output.js:16:38)
33 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1126:14)
34 verbose cwd C:\Sites\AM-API-MDD
35 verbose Windows_NT 10.0.17763
36 verbose node v16.17.0
37 verbose npm  v8.15.0
38 error Unexpected token '.'
39 verbose exit 1
40 timing npm Completed in 87ms
41 verbose code 1
42 error A complete log of this run can be found in:
42 error     C:\Users\tech-cclark\AppData\Local\npm-cache\_logs\2022-09-07T20_47_04_581Z-debug-0.log
```

### Expected behavior

Environment variables loaded from ..\.env
Prisma schema loaded from schema.prisma

✔ Generated Prisma Client (4.3.1 | library) to .\..\node_modules\@prisma\client in 1.12s
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```

### Prisma information

This command works using 
node -v 14.19.0
npm -v 6.14.17

### Environment & setup

Windows Server 2019 Standard v 1809, OS Build 17763.3287
Nodejs -v 16.17.0
npm -v 8.11.0

### Prisma Version

prisma -v 4.3.1

**Comment by nurul3101** [comment]
Hey @ChipClark 👋 
Just to confirm, `npx prisma generate` command works for you in node version `14.19.0` but not in version `16.17.0`. Am I understanding this correctly?

**Comment by ChipClark** [comment]
Correct.


Chip Clark

**Reply by nurul3101** [reply]
Is your repository public? Could you share a minimal reproduction with us so that we can check it on our end?

**Comment by ChipClark** [comment]
I should be able to give you a reproduction later today.


Chip Clark

**Reply by vladi-stevanovic** [reply]
👋 Hey @ChipClark, I wanted to follow up to see if this is still an issue for you? Let us know if you could share a minimal reproduction with us so that we can continue investigating (and open a GH issue to report a bug if needed! ) 🙂

**Comment by ChipClark** [comment]
I am no longer at that company. I have code which can be installed which
might replicate the issue.  Because of my change in jobs this is no longer
an issue for me.

Chip

**Reply by vladi-stevanovic** [reply]
Ok, thank you for letting us know Chip! If you come across any other questions, don't hesitate to let us know!

**Comment by ChipClark** [comment]
Will do, thanks!
Chip Clark

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
